micrometer-observation/src/main/java/io/micrometer/observation/SimpleObservation.java

```diff
@@ -55,6 +55,32 @@ class SimpleObservation implements Observation {
                 .filter(handler -> handler.supportsContext(this.context))
                 .collect(Collectors.toCollection(ArrayDeque::new));
         this.filters = registry.observationConfig().getObservationFilters();
+        List<ObservationConvention<?>> observationConventions = registry.observationConfig().getObservationConventions()
```

Comment:
When using the constructor with the `ObservationConvention<?>` parameter, only the given `ObservationConvention` is applied. In this case, ALL matching conventions are applied. I wonder if this is intended? Why doesn't this code use `io.micrometer.observation.ObservationRegistry.ObservationConfig#getObservationConvention`, which returns the first matching one? I must admit i'm a bit confused, so I can be totally wrong here.